### DISCUSSION TITLE
KeyError 'ganacheLocal'

### DISCUSSION THREAD

**Discussion by wilonweb** [smartcontractkit/full-blockchain-solidity-course-py#1458] [Q&A] [answered]
Hello

when i launch the command `brownie run scripts/deploy.py --network ganache-local` i have a keyError
Do you know what that means ?

`
wilfried@DESKTOP-FQ2D16J:~/smartContract/brownie_fund_me$ brownie run scripts/deploy.py --network ganache-local
Brownie v1.18.1 - Python development framework for Ethereum

BrownieFundMeProject is the active project.
  File "brownie/_cli/__main__.py", line 64, in main
    importlib.import_module(f"brownie._cli.{cmd}").main()
  File "brownie/_cli/run.py", line 45, in main
    network.connect(CONFIG.argv["network"])
  File "brownie/network/main.py", line 31, in connect
    active = CONFIG.set_active_network(network)
  File "brownie/_config.py", line 76, in set_active_network
    network = self.networks[id_].copy()
KeyError: 'ganache-local'
`

**Comment by cromewar** [comment, marked as answer]
run `brownie networks list` and see if you have correctly added ganache-local, if is present and still does not work, please delete the network and add it again.

**Reply by wilonweb** [reply]
Yes ganache-local was not on the network list ..
I add it and now i got a `FileNotFoundError: [Errno 2] No such file or directory: 'ganache'` 
When i launch `brownie run scripts/deploy.py --network ganache-local`
The output is 
```
wilfried@DESKTOP-FQ2D16J:~/smartContract/brownie_fund_me$ brownie run scripts/deploy.py --network ganache-local
Brownie v1.18.1 - Python development framework for Ethereum

BrownieFundMeProject is the active project.
  File "brownie/_cli/__main__.py", line 64, in main
    importlib.import_module(f"brownie._cli.{cmd}").main()
  File "brownie/_cli/run.py", line 45, in main
    network.connect(CONFIG.argv["network"])
  File "brownie/network/main.py", line 50, in connect
    rpc.launch(active["cmd"], **active["cmd_settings"])
  File "brownie/network/rpc/__init__.py", line 75, in launch
    self.process = self.backend.launch(cmd, **kwargs)
  File "brownie/network/rpc/ganache.py", line 69, in launch
    ganache_version = get_ganache_version(cmd_list[0])
  File "brownie/network/rpc/ganache.py", line 110, in get_ganache_version
    ganache_version_proc = psutil.Popen([ganache_executable, "--version"], stdout=PIPE)
  File "psutil/__init__.py", line 1319, in __init__
    self.__subproc = subprocess.Popen(*args, **kwargs)
  File "/usr/lib/python3.8/subprocess.py", line 858, in __init__
    self._execute_child(args, executable, preexec_fn, close_fds,
  File "/usr/lib/python3.8/subprocess.py", line 1704, in _execute_child
    raise child_exception_type(errno_num, err_msg, err_filename)
FileNotFoundError: [Errno 2] No such file or directory: 'ganache'
```

**Reply by cromewar** [reply]
When you are running ganache-local I suppose you are using ganche-ui, so please go to ganache config and change the server to WSL
![image](https://user-images.githubusercontent.com/20199180/166970497-c27de940-944f-4e89-80de-1d29ff3288d1.png)

**Reply by wilonweb** [reply]
I don't know what i did but i have only 0.0.0.0 - All interfaces in the list

![ttyty](https://user-images.githubusercontent.com/10622850/167003711-f919a29f-587f-4f17-8781-cbf7487feb99.png)
![test5](https://user-images.githubusercontent.com/10622850/167003814-233f8ad8-364c-4b5b-adc6-42ec1a6608fc.png)

**Reply by cromewar** [reply]
Common error, just do the following:

1. run windows + R
2. write %APPDATA% 
3. delete the ganache folder
4. Open it again
5. Your problem is solved now! ;)
